pkg/auth/user/util.go

```diff
@@ -25,3 +27,14 @@ func ConvertAttributes(attrMap map[string][]string) []*v1.UserAttribute {
 	})
 	return result
 }
+
+// LogSuccessfulUserLogin logs user attributes in the specified logger instance.
+func LogSuccessfulUserLogin(log *logging.Logger, user *v1.AuthStatus) {
+	loggableUser := user.Clone()
+	// Auth provider config can contain sensitive data(client secret, certificates etc.) so it shouldn't be logged.
+	if loggableUser != nil {
+		loggableUser.AuthProvider = nil
+	}
+	userJSON, _ := jsonutil.ProtoToJSON(loggableUser)
+	log.Warnf("User successfully logged in with user attributes: %s", userJSON)
```

Comment:
I'd suggest to use a structured log entry here instead of adding the json to the message string directly.

Reply:
Do you suggest going over the fields we want to log individually in the structured log:
```
log.Warnw("User successfully logged in with user attributes",
		zap.String("userID", loggableUser.GetUserId()),
         zap.String("username", loggableUser.GetUserInfo().GetUsername())
```
**or** just to add JSON to the log via `Warnw`:
```
log.Warnw("User successfully logged in with user attributes",
		zap.String("userEntry", userJSON))
```
**or** log proto directly via `Warnw`:
```
log.Warnw("User successfully logged in with user attributes",
		zap.Any("userEntry", loggableUser))
```

The problem with the second approach is that it messes up formatting - we would get all escape characters written down:
```
{"userEntry": "{\n  \"userId\": \"UserID\",\n  \"expires\": \"2023-02-20T21:25:09.770338Z\",\n  \"userInfo\": {\n    \"username\": \"UserID\",\n    \"friendlyName\": \"Madonna Slytherin\",\n    \"permissions\": {\n      \"resourceToAccess\": {\n        \"Close Magic Doors\": \"READ_ACCESS\",\n        \"Open Magic Doors\": \"READ_WRITE_ACCESS\"\n      }\n    }\n  },\n  \"userAttributes\": [\n    {\n      \"key\": \"a\",\n      \"values\": [\n        \"b\"\n      ]\n    }\n  ]\n}"}
```

The problem with the third approach is that also messes up formatting + does not support google protobuf timestamps:
```
 {"userEntry": "user_id:\"UserID\" expires:<seconds:1676928078 nanos:435893000 > user_info:<username:\"UserID\" friendly_name:\"Madonna Slytherin\" permissions:<resource_to_access:<key:\"Close Magic Doors\" value:READ_ACCESS > resource_to_access:<key:\"Open Magic Doors\" value:READ_WRITE_ACCESS > > > user_attributes:<key:\"a\" values:\"b\" > "}
```

Reply:
I think it makes sense to select the fields we want to log rather than logging everything except specific elements.
This is less prone to inadvertently logging a sensitive field that could be added to the data in the future.

Reply:
I switched to logging selected fields instead of the entire `v1.AuthStatus`, which should solve this issue.
One good argument that persuaded me is that we want to decouple logged structure from `v1.AuthStatus` as future developers might add a new field with sensitive data.